retrace/glstate_shaders.cpp

```diff
@@ -1049,6 +1049,136 @@ dumpShadersUniforms(StateWriter &writer, Context &context)
+    glGetProgramResourceName(program, programInterface, index, nameLength, nullptr, buffer.data());
+    return {buffer.begin(), buffer.end()};
+}
+
+void dumpShadersStorageBufferBlocks(StateWriter &writer, Context &context,
+                                    GLuint program) {
+    if (!(context.ARB_shader_storage_buffer_object &&
+          context.ARB_program_interface_query))
+        return;
+    GLint numberOfActiveSSBBs = -1;
+    glGetProgramInterfaceiv(program, GL_SHADER_STORAGE_BLOCK,
+                            GL_ACTIVE_RESOURCES, &numberOfActiveSSBBs);
+    //writer.writeIntMember("GL_ACTIVE_RESOURCES", numberOfActiveSSBBs);
+    if (numberOfActiveSSBBs > 0) {
+        for (GLint ssbbResourceIndex = 0;
+             ssbbResourceIndex < numberOfActiveSSBBs; ++ssbbResourceIndex) {
+            GLint numberOfActiveVariables =
+                getProgramResourcei(program, GL_SHADER_STORAGE_BLOCK,
+                                    ssbbResourceIndex, GL_NUM_ACTIVE_VARIABLES);
+            auto ssbbName = getProgramResourceName(
+                program, GL_SHADER_STORAGE_BLOCK, ssbbResourceIndex);
+            auto activeVariables = getProgramResourcei(
+                program, GL_SHADER_STORAGE_BLOCK, ssbbResourceIndex,
+                {GL_ACTIVE_VARIABLES}, numberOfActiveVariables);
+            writer.beginMember(ssbbName);
+            writer.beginObject();
+            const GLint bufferBinding =
+                getProgramResourcei(program, GL_SHADER_STORAGE_BLOCK,
+                                    ssbbResourceIndex, GL_BUFFER_BINDING);
+            writer.writeIntMember("GL_BUFFER_BINDING", bufferBinding);
+            writer.writeIntMember(
+                "GL_BUFFER_DATA_SIZE",
+                getProgramResourcei(program, GL_SHADER_STORAGE_BLOCK,
+                                    ssbbResourceIndex, GL_BUFFER_DATA_SIZE));
+            GLint bufferName = 0;
+            glGetIntegeri_v(GL_SHADER_STORAGE_BUFFER_BINDING, bufferBinding, &bufferName);
+            if(bufferName != 0) {
+               writer.writeIntMember("Buffer", bufferName);
+            }
+            writer.beginMember("activeVariables");
+            writer.beginObject();
+            for (GLint variableIndex : activeVariables) {
+                auto variableName = getProgramResourceName(
+                    program, GL_BUFFER_VARIABLE, variableIndex);
+                writer.beginMember(variableName);
+                writer.beginObject();
+
+#define DUMP_BUFFER_VARIABLE_INT(NAME)                                         \
+    do {                                                                       \
+        int value = getProgramResourcei(program, GL_BUFFER_VARIABLE,           \
+                                        variableIndex, NAME);                  \
+        writer.writeIntMember(#NAME, value);                                   \
+    } while (false)
+
+                DUMP_BUFFER_VARIABLE_INT(GL_TYPE);
```

Comment:
Like my other comment, please use GLenum table, plus enumToString to obtain C strings.